src/openagi/actions/tools/huggingface.py

```diff
@@ -0,0 +1,29 @@
+from openagi.actions.base import ConfigurableAction
+from pydantic import Field
+import json
+
+try:
+    from huggingface_hub import HfApi
+except ImportError:
+    raise Exception("Install huggingface_hub with cmd `pip install huggingface`")
+
+class HuggingFaceTool(ConfigurableAction):
+    """
+    Tool for exploring Hugging Face models and datasets.
+    """
+    query: str = Field(..., description="Dataset or model name to search on Hugging Face.")
+
+    def execute(self) -> str:
+        api = HfApi()
+        results_dict = {}
+
+        models = api.list_models(search=self.query, limit=15)
+        datasets = api.list_datasets(search=self.query, limit=15)
+
+        for model in models:
```

Comment:
can you also extract other details as well. just model ID is very limited context given to the LLM